### PR TITLE
Convert worksapce schematics for nx

### LINKED ISSUE #1488: Move schematics to workspace schematics.
# Feature Request

## Summary

Currently, our schematics have to be built first and then run out of the dist folder which is quite cumbersome. Nx provides a way of directly running them without building them first. And we already have it in our npm scripts with `npm run workspace-schematic`

## Feature Description

If you want to create a new example or whatever you have to know the schematics, then it needs to be built and you have to know how to execute it. Like this: 

```bash
ng build workspace
ng g ./dist/libs/workspace:dt-example --name=type-of-example --component=my-component
```

The nx workspace schematics are shipped with nx and you can run `nx workspace-schematic` and get a help page already

```
➜ npm run workspace-schemati
$ node --max_old_space_size=8192 ./node_modules/.bin/nx workspace-schematic
Positionals:
  name  The name of your schematic`                          [string] [required]

Options:
  --help             Show help                                         [boolean]
  --version          Show version number                               [boolean]
  --list-schematics  List the available workspace-schematics           [boolean]
```

The consumer could then run:

```
➜ npm run workspace-schematic --list-schematics
$ nx workspace-schematic --list-schematics
component
component-dev
component-e2e
component-example
```

You can then invoke it with: `yarn workspace-schematic component my-component-name --dry-run`. 
Easy to use no need to build anything before as it uses a ts-node under the hood.

Furthermore, workspace schematics can be executed directly via the NxPlugin in VSCode:
image

image

And remove the old build step -> move to tools folder

### PR DESCRIPTION
**Pull request by bardosmisi**
### <strong>Pull Request</strong>

Convert schematics to NX workspace schematics.

#### Type of PR

Feature (non-breaking change which adds functionality)

Closes #1488

### CODE REVIEW

File: tools/schematics/dt-component-dev/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Dev component schematics
+
+This schematics generates the component for the barista dev environment app. The
```
Comment (rowa-audil) [inline tools/schematics/dt-component-dev/README.md:3]: NIT: 
```suggestion
This schematic generates the component for the barista dev environment app. The
```

File: tools/schematics/dt-component-e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# E2E Test component schematics
+
+This schematics generates the component for barista e2e tests. The only needed
```
Comment (rowa-audil) [inline tools/schematics/dt-component-e2e/README.md:3]:
```suggestion
This schematic generates the component for barista e2e tests. The only needed
```

File: tools/schematics/dt-component-dev/README.md (hunk 1 above)
Comment (rowa-audil) [inline tools/schematics/dt-component-dev/README.md:3]:
```suggestion
This schematic generates the component for the barista dev environment app. The
```

File: tools/schematics/dt-component-e2e/README.md (hunk 2 above)
Comment (rowa-audil) [inline tools/schematics/dt-component-e2e/README.md:3]:
```suggestion
This schematic generates the component for barista e2e tests. The only needed
```

File: tools/schematics/dt-component-example/README.md [hunk 3]
```diff
@@ -13,9 +13,5 @@ hand.
 
 ## Usage
 
-- Please run: `ng build workspace` and after that,
-  `ng g ./dist/libs/workspace:dt-example --name={name of your example} --component={name of your component}`
-
-### Testing
-
-The schematics can be tested with the `ng test workspace` command.
+- Please run: ``nx workspace-schematic dt-component-dev {name of your example}
```
Comment (rowa-audil) [inline tools/schematics/dt-component-example/README.md:16]: L3 and L8: schematic instead of plural

File: tools/tsconfig tryouts/tsconfig.spec.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  // "extends": "./tsconfig.json",
```
Comment (rowa-audil) [inline tools/tsconfig tryouts/tsconfig.spec.json:2]: Do the slashes not throw an error? 🤔

**Review by rowa-audil — COMMENTED** [review]
NITs but overall LGTM! 🔧 

So many renames 😄

**Review by rowa-audil — APPROVED** [review]

**Review by peter-affenzeller — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -115,7 +116,7 @@
     "@nrwl/linter": "^10.0.7",
     "@nrwl/node": "^10.0.7",
     "@nrwl/nx-plugin": "^10.0.7",
-    "@nrwl/workspace": "^10.0.7",
+    "@nrwl/workspace": "^10.2.1",
```
Comment (lukasholzer) [inline package.json:119]: please revert the changes in the package-lock.json and in the `package.json`. You have to have all `@nrwl/` dependencies on the same version.

Comment (lukasholzer) [reply package.json:119]: I would bump the version of nrwl depenendcies in a seperate PR

File: tools/schematics/dt-component-e2e/index.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ import {
   template,
   Tree,
   url,
-} from '@angular-devkit/schematics';
+} from '@nrwl/workspace/node_modules/@angular-devkit/schematics';
```
Comment (lukasholzer) [inline tools/schematics/dt-component-e2e/index.ts:26]: wrong import path never import from node_modules revert this change

File: tools/schematics/dt-component-example/example-tools.utils.ts [hunk 7]
```diff
@@ -14,7 +14,10 @@
  * limitations under the License.
  */
 
-import { Rule, Tree } from '@angular-devkit/schematics';
+import {
+  Rule,
+  Tree,
+} from '@nrwl/workspace/node_modules/@angular-devkit/schematics';
```
Comment (lukasholzer) [inline tools/schematics/dt-component-example/example-tools.utils.ts:20]: never import from node_modules revert this change

File: tools/schematics/dt-component-example/index.ts [hunk 8]
```diff
@@ -14,37 +14,37 @@
...
   Tree,
   url,
-  noop,
-} from '@angular-devkit/schematics';
-import { formatFiles } from '@nrwl/workspace';
-import { DtComponentExampleOptions } from './schema';
-import { normalize } from '@angular-devkit/core';
-import { LICENSE_HEADER } from '../../utils/common-utils';
+  SchematicContext,
+} from '@nrwl/workspace/node_modules/@angular-devkit/schematics';
```
Comment (lukasholzer) [inline tools/schematics/dt-component-example/index.ts:30]: never import from node_modules revert this change

File: tools/schematics/dt-component-dev/index.ts [hunk 9]
```diff
@@ -23,14 +23,13 @@ import {
   template,
   Tree,
   url,
-} from '@angular-devkit/schematics';
+} from '@nrwl/workspace/node_modules/@angular-devkit/schematics';
```
Comment (lukasholzer) [inline tools/schematics/dt-component-dev/index.ts:26]: revert this change – never import from sub node_modules

**Review by lukasholzer — CHANGES_REQUESTED** [review]

File: tools/schematics/dt-component-dev/index.ts (hunk 9 above)
Comment (bardosmisi) [reply tools/schematics/dt-component-dev/index.ts:26]: Yes, I guess I needed to do that because of the version difference in the package.json. The fix is coming up.

Comment (bardosmisi) [reply tools/schematics/dt-component-dev/index.ts]: I removed all sub-node_module imports.

File: tools/schematics/dt-component-example/index.ts (hunk 8 above)
Comment (bardosmisi) [reply tools/schematics/dt-component-example/index.ts]: I removed all sub-node_module imports.

File: tools/schematics/dt-component-example/example-tools.utils.ts (hunk 7 above)
Comment (bardosmisi) [reply tools/schematics/dt-component-example/example-tools.utils.ts]: I removed all sub-node_module imports.

File: tools/schematics/dt-component-e2e/index.ts (hunk 6 above)
Comment (bardosmisi) [reply tools/schematics/dt-component-e2e/index.ts]: I removed all sub-node_module imports.

File: package.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dynatrace-oss/barista",
-  "version": "8.0.3",
+  "version": "8.1.0",
```
Comment (bardosmisi) [inline package.json:3]: This change should disappear after rebase.

File: package.json (hunk 5 above)
Comment (bardosmisi) [reply package.json]: Thank you for the thorough review, I think now there are no changes to the current package.json in the pr.

**Review by rowa-audil — DISMISSED** [review]

**Review by lukasholzer — DISMISSED** [review]
LGTM please rebase your changes and squash to one commit

**Review by rowa-audil — DISMISSED** [review]

**Review by peter-affenzeller — APPROVED** [review]

**Comment by lukasholzer** [comment]
please rename to a chore commit as it is not a consumer facing feature.

**Review by rowa-audil — APPROVED** [review]

**Review by lukasholzer — APPROVED** [review]
